src/Config.js

```diff
@@ -566,9 +682,16 @@ export class Config {
     if (databaseOptions == undefined) {
       return;
     }
+
     if (Object.prototype.toString.call(databaseOptions) !== '[object Object]') {
       throw `databaseOptions must be an object`;
     }
+    Config.validateConfigKeyNames(
```

Comment:
As you can see with some of the keys the types for them need to be validated. For eg. as you can see: rateLimit must be an array or object, and all these checks were already done in the respective validate functions, that is why I decided to split the tests rather than at one position.
But your point makes sense as well. I'll look into it.

Reply:
So, I looked into it. The problem I see is that during execution I won't have the information of which key belongs to which interface and so, it will be difficult to match key with the correct object. I'm new to flowJS so maybe I'm missing something. Please do let me know if there is a way.

Reply:
In the constructor `constructor(options: ParseServerOptions)` you get all the options, for example:

```
{
  serverURL: "http://localhost:8378/1",
  appId: "test",
  javascriptKey: "test",
  dotNetKey: "windows",
  clientKey: "client",
  restAPIKey: "rest",
  webhookKey: "hook",
  masterKey: "test",
  maintenanceKey: "testing",
  readOnlyMasterKey: "read-only-test",
  fileKey: "test",
  directAccess: true,
  silent: true,
  logLevel: undefined,
  fileUpload: {
    enableForPublic: true,
    enableForAnonymousUser: true,
    enableForAuthenticatedUser: true,
  },
  push: {
    android: {
      senderId: "yolo",
      apiKey: "yolo",
    },
  },
  allowClientClassCreation: true,
  mountPath: "/1",
  port: 8378,
}
```

Then could you just import the definitions with `const definitions = require('../lib/Options/Definitions.js');` and go through all the exports to create a list of names and types? Maybe the type currently can ony be inferred from the `action` key. The kind of parser gives you an idea about the type. But mybe you could extend `resources/buildConfigDefinitions.js` to add an explicit `type` property to the generated output in `Definitions.js`?

Then you just compare the two.

Reply:
Great, I understand the idea. 

The issue you mentioned talks about grouping various config options, but as long as we have definitions file out solution should work just fine even after re-organizing the config variables, don't you think?
If so, should I start updating to include this?

Reply:
Yes, maybe you could try to change it according to https://github.com/parse-community/parse-server/pull/8987#discussion_r1526477303, and if you find it's too complicated, we could stay with your current approach.

Reply:
Great!

Reply:
Let's keep it this way only, if that's alright then. 

The thing is the value given by user might not be the expected value type so, for that I'll have to write validation that are already done in specific functions, which would be redundant. I was able to pull interface types and put them in definition files as you suggested. But I remember there is one key that can be array of an object, other keys which are an only object and so on which can lead to me repeating a lot of code for type validation.

Reply:
That means we need to update the documentation, as contributors will have an additional step to consider, in order to get new Parse Server options validated, right? I suggest you add the change in this PR to the CONTRIBUTING.md file under "Parse Server Options" and let's see what's required. If it turns out to be too complex, we can reconsider whether we really want to add the feature in its current form.

Reply:
I think it looks good.